spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2300,6 +2300,12 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
             expr.children(1),
             inputs,
             (builder, binaryExpr) => builder.setArrayAppend(binaryExpr))
+        case ArraysOverlap(leftArrayExpr, rightArrayExpr) =>
```

Comment:
Do we support this for all input data types?

Reply:
Because the PR only contains basic tests, could you add a check to enable this expression only if `CometConf.COMET_CAST_ALLOW_INCOMPATIBLE` is enabled? We can remove this check in a future PR that adds comprehensive tests and demonstrates that we have Spark-compatible behavior for all supported data types.`

Reply:
Yes, makes sense and `COMET_CAST_ALLOW_INCOMPATIBLE` check has been added.